crates/bevy_tasks/src/task_pool.rs

```diff
@@ -532,11 +461,17 @@ impl TaskPool {
     ///
     /// If the provided future is non-`Send`, [`TaskPool::spawn_local`] should
     /// be used instead.
-    pub fn spawn<T>(&self, future: impl Future<Output = T> + Send + 'static) -> Task<T>
+    pub fn spawn_async<T>(&self, future: impl Future<Output = T> + Send + 'static) -> Task<T>
     where
         T: Send + 'static,
     {
-        Task::new(self.executor.spawn(future))
+        let (runnable, task) = async_task::spawn(future, |runnable: Runnable| {
+            rayon_core::spawn(move || {
```

Comment:
Note that `rayon_core::spawn` spawns the task on whatever rayon context is the current one in the thread it is called in. Most notably, the `async_task` closure may be called on a different thread than the one the task was spawned in (e.g. if the `Future`'s `Waker` was moved into a different thread), in which case this would likely initialize a different global rayon's context, with a different thread pool.